okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.java

```diff
@@ -56,6 +57,12 @@
    */
   private static final long MAX_QUEUE_SIZE = 16 * 1024 * 1024; // 16 MiB.
 
+  /**
+   * The maximum amount of time after the client calls {@link #close} to wait for a graceful
+   * shutdown. If the server doesn't respond the websocket will be canceled.
+   */
+  private static final long CANCEL_AFTER_CLOSE_MILLIS = 60 * 1000;
```

Comment:
Use the connect timeout? [Dis]connect timeout, am I right? Otherwise even read timeout seems appropriate.